test/unittest/tests/block/test_block_addr.cpp

```diff
@@ -0,0 +1,174 @@
+        CHECK(unpacked_size == 0);
+        CHECK(unpacked_checksum == 0);
+    }
+}
+
+// Test the block manager's pack function.
+static void
+test_pack_addr_cookie(uint8_t *pp, WT_BLOCK *block, size_t *addr_size, wt_off_t pack_offset,
+  uint32_t pack_size, uint32_t pack_checksum)
+{
+    const uint8_t *begin = (const uint8_t *)pp;
+    REQUIRE(__wt_block_addr_pack(
+              block, &pp, WT_TIERED_OBJECTID_NONE, pack_offset, pack_size, pack_checksum) == 0);
+    *addr_size = WT_PTRDIFF(pp, begin);
+    unpack_addr_cookie_and_check(begin, block->allocsize, pack_offset, pack_size, pack_checksum);
+}
+
+// Test the block manager's unpack function.
+static void
+test_unpack_addr_cookie(const uint8_t *begin, WT_BLOCK *block, size_t addr_size,
+  wt_off_t pack_offset, uint32_t pack_size, uint32_t pack_checksum)
+{
+    uint32_t checksum, obj_id, size;
+    wt_off_t offset;
+    REQUIRE(__wt_block_addr_unpack(
+              NULL, block, begin, addr_size, &obj_id, &offset, &size, &checksum) == 0);
+    if (size != 0) {
+        CHECK(offset == pack_offset);
+        CHECK(size == pack_size);
+        CHECK(checksum == pack_checksum);
+    } else {
+        CHECK(offset == 0);
+        CHECK(size == 0);
+        CHECK(checksum == 0);
+    }
+}
+
+static void
+test_pack_and_unpack_addr_cookie(
+  WT_BLOCK *block, wt_off_t pack_offset, uint32_t pack_size, uint32_t pack_checksum)
+{
+    uint8_t p[WT_BTREE_MAX_ADDR_COOKIE], *pp;
+    pp = p;
+
+    const uint8_t *begin = (const uint8_t *)pp;
+    size_t addr_size;
+
+    test_pack_addr_cookie(pp, block, &addr_size, pack_offset, pack_size, pack_checksum);
+    test_unpack_addr_cookie(begin, block, addr_size, pack_offset, pack_size, pack_checksum);
+}
+
+static void
+test_pack_and_unpack_addr_cookie_manual(
+  WT_BLOCK *block, std::vector<int64_t> cookie_vals, std::vector<uint8_t> &expected_packed_vals)
+{
+    std::vector<uint8_t> packed(24, 0);
+    uint8_t *p = packed.data();
+    const uint8_t *begin = (const uint8_t *)p;
+    REQUIRE(__wt_block_addr_pack(block, &p, WT_TIERED_OBJECTID_NONE, (wt_off_t)cookie_vals[0],
+              (uint32_t)cookie_vals[1], (uint32_t)cookie_vals[2]) == 0);
+    CHECK(packed[0] == expected_packed_vals[0]);
+    CHECK(packed[1] == expected_packed_vals[1]);
+    CHECK(packed[2] == expected_packed_vals[2]);
+
+    unpack_addr_cookie_and_check(begin, block->allocsize, (wt_off_t)cookie_vals[0],
+      (uint32_t)cookie_vals[1], (uint32_t)cookie_vals[2]);
+}
+
+TEST_CASE("Block addr pack and unpack", "[block_addr]")
+{
+    WT_BLOCK b;
+    WT_BM bm, *bmp;
+
+    b.allocsize = 1;
+    bmp = &bm;
+    bmp->block = &b;
+
+    // Test the block manager's pack function with an address cookie containing all zero fields.
+    SECTION("Pack and unpack address cookie 1")
+    {
+        uint32_t pack_checksum = 0, pack_size = 0;
+        wt_off_t pack_offset = 0;
+
+        test_pack_and_unpack_addr_cookie(bmp->block, pack_offset, pack_size, pack_checksum);
```

Comment:
Sorry I didn't catch this earlier, i would just place the numbers into here rather than creating a variable. Althought we do lose readability in this case. I want to see what other people think of this.

Reply:
I agree that we can add them directly to the function call, but the comment above the function call should read:
// (0,0,0) -> (offset, size, checksum)

And so on for the other ones.